response_operations_ui/views/respondents.py

```diff
@@ -14,22 +17,78 @@
                           static_folder='static', template_folder='templates')
 
 
-@respondent_bp.route('/', methods=['GET', 'POST'])
+@respondent_bp.route('/', methods=['GET'])
+@login_required
+def respondent_home():
+    return render_template('respondent-search/respondent-search.html',
+                           form=RespondentSearchForm(),
+                           breadcrumbs=[{"title": "Respondents"}])
+
+
+@respondent_bp.route('/search', methods=['POST'])
+@login_required
+def search_redirect():
+    form = RespondentSearchForm()
+    form_valid = form.validate()
+
+    if not form_valid:
+        flash('At least one input should be filled')
+        return redirect(url_for('respondent_bp.respondent_home'))
+
+    return redirect(url_for('respondent_bp.respondent_search',
+                            email_address=form.email_address.data or '',
+                            first_name=form.first_name.data or '',
+                            last_name=form.last_name.data or '',
+                            page=request.args.get('page', 1)))
+
+
+@respondent_bp.route('/search')
+@respondent_bp.route('/search/', methods=['GET'])
 @login_required
 def respondent_search():
-    form = SearchForm()
-    breadcrumbs = [{"title": "Respondents"}]
-    response = None
+    breadcrumbs = [{"title": "Respondents"}, {"title": "Search"}]
+
+    first_name = request.values.get('first_name', '')
+    last_name = request.values.get('last_name', '')
+    email_address = request.values.get('email_address', '')
+    page = request.values.get('page', '')
+
+    form = RespondentSearchForm()
+
+    form.first_name.data = first_name
+    form.last_name.data = last_name
+    form.email_address.data = email_address
+
+    party_response = party_controller.search_respondents(first_name, last_name, email_address, page)
+
+    respondents = party_response.get('data', [])
+    total_respondents_available = party_response.get('total', 0)
+
+    filtered_respondents = filter_respondents(respondents)
+
+    RESULTS_PER_PAGE = app.config["PARTY_RESPONDENTS_PER_PAGE"]
```

Comment:
More minor stuff, this shouldn't be uppercased as no other variables we use are uppercased (even if the config item is)